### PR TITLE
Improve contribution documentation

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Improve contribution documentation (#572)

- Introduce a `./run-full-build.sh` script.
- Explicitly mention that users should run this script before opening a pull
  request.
- Emphasize that many build warnings can be resolved automatically.
- Introduce a `SECURITY.md` file as suggested by GitHub.
```

Context of this PR: I've registered Error Prone Support with the [OpenSSF Best Practices Badge Program](https://bestpractices.coreinfrastructure.org/en/projects/7199), and these changes increase our compliance. (Other PRs to improve compliance may follow.)

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+use the latest release where feasible.
+
+## Reporting a vulnerability
+
+To report a vulnerability, please visit the [security
+advisories][security-advisories] page an click _Report a vulnerability_. We
+will take such reports seriously and work with you to resolve the issue in a
+timely manner.
+
+[security-advisories]: https://github.com/PicnicSupermarket/error-prone-support/security/advisories
```
Comment (japborst) [inline SECURITY.md:22]: Did not know this existed. Nice!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -48,6 +48,10 @@ be accepted. When in doubt, make sure to first raise an
 
 To the extent possible, the pull request process guards our coding guidelines.
 Some pointers:
+- Make sure that the `mvn clean install` build fully passes, ideally before
+  opening a pull request. See the [development
```
Comment (japborst) [inline CONTRIBUTING.md:52]: Ideally? Or should we make this more mandatory

```suggestion
- Before opening a Pull Request, make sure that the `mvn clean install` 
build fully passes. See the [development
```

Comment (Stephan202) [reply CONTRIBUTING.md]: I was actually thinking about this last night, and realized that if people get stuck, this requirement might discourage them from contributing at all. I proposed a new sentence; PTAL :)

**Review by japborst — APPROVED** [review]

**Comment by japborst** [comment]
FYI there's a typo on https://bestpractices.coreinfrastructure.org/en/projects/7199. `s/Errorc/Error`

**Comment by Stephan202** [comment]
:facepalm: Fixed!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -58,6 +66,10 @@ Some pointers:
   checks][error-prone-criteria]. Most guidelines described there apply to this
   project as well, except that this project _does_ focus quite heavy on style
   enforcement. But that just makes the previous point doubly important.
+- Make sure that there are no violations of the newly introduced checks in the
+  codebase itself by running the [self check][error-prone-support-self-check].
+  See the [development instructions][error-prone-support-developing] for extra
+  context.
```
Comment (Stephan202) [inline CONTRIBUTING.md:72]: This mostly repeats the text above. The idea behind referencing the `error-prone-support-developing` was that we don't need to repeat the self check description.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -66,6 +78,8 @@ Some pointers:
   sneak in unrelated changes; instead just open more than one pull request 😉.
 
 [error-prone-criteria]: https://errorprone.info/docs/criteria
+[error-prone-support-developing]: https://github.com/PicnicSupermarket/error-prone-support/tree/master#-developing-error-prone-support
 [error-prone-support-issues]: https://github.com/PicnicSupermarket/error-prone-support/issues
 [error-prone-support-mutation-tests]: https://github.com/PicnicSupermarket/error-prone-support/blob/master/run-mutation-tests.sh
 [error-prone-support-pulls]: https://github.com/PicnicSupermarket/error-prone-support/pulls
+[error-prone-support-self-check]: https://github.com/PicnicSupermarket/error-prone-support/blob/master/apply-error-prone-suggestions.sh
```
Comment (Stephan202) [inline CONTRIBUTING.md:85]: This is the auto-fix script, not a full self check. (The details are subtle, which is why I was thinking to only reference the other section.)

**Review by Stephan202 — COMMENTED** [review]
I think we can do better here, e.g. by checking in the `build-all.sh` script I use locally. But likely we should more generally refactor this information so that it's not spread across two files. That seems like a larger task, though...

**Comment by rickie** [comment]
I'm not sure everyone reads till the end in the `error-prone-support-developing` section as the explanation is below "some other commands one may find relevant". We could also reword that to say something along the lines of "important/helpful to run before opening a PR". That would highlight the actual importance IMO.

**Review by rickie — APPROVED** [review]
💯 good stuff! Really nice improvements :).

**Comment by rickie** [comment]
Will rebase and merge this.

**Comment by rickie** [comment]
Tweaked suggested commit message PTAL.

**Comment by Stephan202** [comment]
Good one; made some changes :)

**Comment by rickie** [comment]
Okay I indeed should've done better here 😅. Thanks! Will merge.

**Comment by rickie** [comment]
(Dropped one tiny word, something that is introduced is always new right 😉, and the last entry doesn't have "new").

**Pull request merged**